src/open_company_web/components/growth/growth.cljs

```diff
@@ -22,25 +23,36 @@
 (defn start-editing-cb [owner data]
   (om/set-state! owner :editing true))
 
+(defn start-data-editing-cb [owner data]
+  (om/set-state! owner :data-editing true))
+
 (defn cancel-cb [owner data]
-  (let [section-data (:section-data data)
-        notes-data (:notes section-data)]
-    (om/set-state! owner :title (:title section-data))
-    (om/set-state! owner :growth-data (:data section-data))
-    (om/set-state! owner :growth-metrics (:metrics section-data))
-    (om/set-state! owner :notes-body (:body notes-data))
-    (om/set-state! owner :editing false)))
+  (if (om/get-state owner :oc-editing)
+    ; remove an unsaved section
+    (section-utils/remove-section (:section data))
+    ; cancel editing
+    (let [section-data (:section-data data)
+          notes-data (:notes section-data)]
+      (om/set-state! owner :title (:title section-data))
+      (om/set-state! owner :growth-data (:data section-data))
+      (om/set-state! owner :growth-metrics (:metrics section-data))
+      (om/set-state! owner :notes-body (:body notes-data))
+      (om/set-state! owner :editing false)
+      (om/set-state! owner :data-editing false))))
+
+(defn has-data-changes [owner data]
+  (let [section-data (:section-data data)]
+    (or (not= (:data section-data) (om/get-state owner :section-data))
+        (not= (:metrics section-data) (om/get-state owner :growth-metrics)))))
 
 (defn has-changes [owner data]
   (let [section-data (:section-data data)
         notes-data (:notes-data section-data)]
     (or (not= (:title section-data) (om/get-state owner :title))
-        (not= (:data section-data) (om/get-state owner :section-data))
-        (not= (:metrics section-data) (om/get-state owner :growth-metrics))
         (not= (:body notes-data) (om/get-state owner :notes-body)))))
 
 (defn cancel-if-needed-cb [owner data]
-  (when (not (has-changes owner data))
+  (when (and (not (has-changes owner data)) (not (has-data-changes owner data)))
```

Comment:
i don't here because this is hard to read, i prefer the form:
```
(when (and (not (has-changes ...) (not (has-data-changes ...))
```
since it says when there are not changed, using the or is not so direct to understand..